src/app/dialog/share-view/share-view-dialog.component.ts

```diff
@@ -65,12 +72,20 @@ export class ShareViewDialogComponent implements OnInit, OnDestroy {
   public ngOnInit() {
     this.subscribeToView();
     this.subscribeData();
+    this.parseViewShareUrl();
   }
 
   public ngOnDestroy() {
     this.subscriptions.unsubscribe();
   }
 
+  public copyToClipboard() {
+    document.addEventListener('copy', this.handler);
+    setTimeout(() => {
+      document.execCommand('copy');
+    });
```

Comment:
What about creating `ClipboardService` with a single `copy()` method we can reuse throughout the app and change the implementation later if we decide to use [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API)?

Reply:
But current solution use handler which depends on public property from component. So are you sure to extract this code to `ClipboardService` ?

Reply:
Ok I found solution.